front/src/TrackStatistics/TrackStatistics.js

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+
+import './TrackStatistics.css';
+
+// Material-UI components
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+
+// Import localization
+import { trackStatistics as texts } from '../texts/texts.json';
+
+export function TrackStatistics(props) {
+  const [tracks, setTracks] = useState([]);
+  const [dialogOpen, setDialogOpen] = React.useState(false);
+  const handleStats = () => setDialogOpen(true);
+  const handleClose = () => setDialogOpen(false);
+  const lang = localStorage.getItem('language');
+
+  useEffect(() => {
+    setTracks(props.tracks);
+  }, [props.tracks]);
+
+  // Function for raising or lowering number of visitors in a given track via buttons.
+  // Also checks that the number of visitors will never be less than 0.
+  // Operator is "inc" or "dec"
+  const increment = (trackId, operator) => {
+    setTracks(tracks.map((track) => {
+      if (track.id === parseInt(trackId)) {
+        let newVisitors = track.scheduled.visitors + 1;
+        if (operator === 'dec') {
+          newVisitors = track.scheduled.visitors - 1;
+        }
+        if (operator !== 'dec' || track.scheduled.visitors !== 0) {
+          return {
+            ...track,
+            scheduled: {
+              ...track.scheduled,
+              visitors: newVisitors,
+            },
+          };
+        }
+      }
+      return track;
```

Comment:
Otherwise a nice working solution, but is there a possibility to only modify the data that needs to be modified instead of creating the whole array again?